vizro-mcp/src/vizro_mcp/server.py

```diff
@@ -0,0 +1,299 @@
+                return {"success": False, "error": f"Failed to fetch file: {e!s}"}
+
+        # Check if input is a valid local file
+        path = Path(path_or_url)
+        if path.exists():
+            # Consistent options for both local and remote files
+            df = pd.read_csv(path, on_bad_lines="warn", low_memory=False)
+            return {"success": True, "data": get_dataframe_info(df, path)}
+
+        else:
+            return {
+                "success": False,
+                "error": f"Invalid input: '{path_or_url}' is neither a valid local file nor a GitHub URL",
+            }
+
+    except pd.errors.ParserError as e:
+        # Handle CSV parsing errors specifically
+        return {"success": False, "error": f"Error parsing CSV file: {e!s}"}
+    except requests.exceptions.RequestException as e:
+        # Handle network-related errors
+        return {"success": False, "error": f"Network error when fetching file: {e!s}"}
+
+
+@mcp.prompt(
+    name="create_EDA_dashboard",
+    description="Prompt template for creating an EDA dashboard based on one CSV dataset",
+)
+def create_EDA_dashboard(
+    file_path_or_url: str,
+) -> str:
+    """Prompt template for creating an EDA dashboard based on one CSV dataset."""
+    content = f"""
+Create an EDA dashboard based on the following dataset:{file_path_or_url}. Proceed as follows:
+1. Analyze the data using the load_and_analyze_csv tool first, passing the file path or github url {file_path_or_url}
+    to the tool.
+2. Create a dashboard with 3 pages:
+    - Page 1: Summary of the dataset using the Card component and the dataset itself using the plain AgGrid component.
+    - Page 2: Visualizing the distribution of all numeric columns using the Graph component with a histogram.
+        - use a Parameter that targets the Graph component and the x argument, and you can select the column to
+            be displayed
+        - IMPORTANT:remember that you target the chart like: <graph_id>.x and NOT <graph_id>.figure.x
+        - do not use any color schemes etc.
+        - add filters for all categorical columns
+    - Page 3: Visualizing the correlation between all numeric columns using the Graph component with a scatter plot.
+    - Page 4: Two interesting charts side by side, use the Graph component for this. Make sure they look goog
+        but do not try something beyond the scope of plotly express
+            """
+    return content
+
+
+################# Chart functionality - not sure if I should include this in the MCP server
+
+
+@mcp.tool(name="get_validated_chart_code", description="Validates code created for a chart")
+def get_validated_chart_code(chart_plan: dict[str, Any]) -> str:
+    """Validate the chart code created by the user."""
+    try:
+        chart_plan_obj = ChartPlan(**chart_plan)
+        return chart_plan_obj.model_dump_json()
+    except ValidationError as e:
+        return json.dumps({"error": f"Validation Error: {e!s}"})
```

Comment:
If this is being `json.dump`ed then one of the alternative things like `e.errors()` is probably directly what you want.